api/filesys/fsnode_test.go

```diff
@@ -192,6 +206,54 @@ type pathCase struct {
 	errStr string
 }
 
+func TestFileOps(t *testing.T) {
+	const path = "foo.txt"
+	content := strings.Repeat("longest content", 100)
+
+	fs := MakeFsInMemory()
+	f, err := fs.Create(path)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if _, err := fs.Open(path); err == nil {
+		t.Fatalf("expected already opened error, got nil")
+	}
+	if _, err := fmt.Fprintln(f, content); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if err := f.Close(); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if err := f.Close(); err == nil {
+		t.Fatalf("expected already closed error, got nil")
+	}
+
+	f, err = fs.Open(path)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	defer f.Close()
+
+	buf := make([]byte, 3)
+	if n, err := f.Read(buf); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	} else if n != len(buf) {
+		t.Fatalf("expected reading 3 bytes, read %d", n)
+	}
+	if string(buf) != "lon" {
+		t.Fatalf("expected reading 'lon', got '%s'", buf)
+	}
+	if n, err := f.Read(buf); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	} else if n != len(buf) {
+		t.Fatalf("expected reading 3 bytes, read %d", n)
+	}
+	if string(buf) != "ges" {
+		t.Fatalf("expected reading 'lon', got '%s'", buf)
```

Comment:
```suggestion
		t.Fatalf("expected reading 'ges', got '%s'", buf)
```

Reply:
I changed the tests to be simpler/better